lib/AST/ASTPrinter.cpp

```diff
@@ -2011,6 +2026,27 @@ void PrintAST::visitExtensionDecl(ExtensionDecl *decl) {
     printExtension(decl);
 }
 
+/// Determines if we should print the text of the initializer expression.
+/// There's a very narrow case when we would:
+/// If the decl:
+///   - has an initializer expression
+///   - is @usableFromInline
+/// If the parent type:
+///   - is @_fixed_layout (or effectively fixed-layout)
+///   - has at least one inlinable initializer in itself or any of its
+///     extensions.
+static bool shouldPrintInitializerExpr(const VarDecl *vd) {
+  if (!vd->getAttrs().hasAttribute<HasInitialValueAttr>() &&
+      !vd->getAttrs().hasAttribute<UsableFromInlineAttr>())
+    return false;
+  auto parent = dyn_cast<NominalTypeDecl>(vd->getDeclContext());
+  if (!parent) return false;
+  if (parent->isFormallyResilient())
```

Comment:
@slavapestov This seemed like the right check to do, but is it? I'm trying to see if the type is effectively fixed-layout. Should I be guarding this with a check if the module is resilient?